vortex-array/src/array/extension/compute/mask.rs

```
@@ -0,0 +1,47 @@
+use std::sync::Arc;
+
+use vortex_dtype::{DType, Nullability};
+use vortex_error::{vortex_bail, VortexResult};
+
+use crate::array::extension::ExtensionArray;
+use crate::array::ExtensionEncoding;
+use crate::compute::{mask, FilterMask, MaskFn};
+use crate::{ArrayDType as _, ArrayData, IntoArrayData};
+
+impl MaskFn<ExtensionArray> for ExtensionEncoding {
+    fn mask(&self, array: &ExtensionArray, filter_mask: FilterMask) -> VortexResult<ArrayData> {
+        let DType::Extension(ext_dtype) = array.dtype() else {
+            vortex_bail!("extension array must have extension dtype");
```

Doesn't ExtensionArray have a `ext_dtype()` function? If not, it should. And it should panic, rather than return a result.

---

I reverted the extension changes in  757d1b823. I will revisit when #1965 (casting) merges.